server/base/src/main/java/org/apache/accumulo/server/fs/PreferredVolumeChooser.java

```diff
@@ -21,84 +21,90 @@
   private static final Logger log = LoggerFactory.getLogger(PreferredVolumeChooser.class);
 
-  /**
-   * This should match {@link Property#TABLE_ARBITRARY_PROP_PREFIX}
-   */
-  public static final String PREFERRED_VOLUMES_CUSTOM_KEY = "table.custom.preferredVolumes";
-  // TODO ACCUMULO-3417 replace this with the ability to retrieve by String key.
-  private static final Predicate<String> PREFERRED_VOLUMES_FILTER = key -> PREFERRED_VOLUMES_CUSTOM_KEY.equals(key);
+  public static final String PREFERRED_VOLUMES_CUSTOM_KEY = Property.TABLE_ARBITRARY_PROP_PREFIX + "preferredVolumes";
 
   @SuppressWarnings("unchecked")
   private final Map<String,Set<String>> parsedPreferredVolumes = Collections.synchronizedMap(new LRUMap(1000));
   // TODO has to be lazily initialized currently because of the reliance on HdfsZooInstance. see ACCUMULO-3411
   private volatile ServerConfigurationFactory serverConfs;
 
   @Override
-  public String choose(VolumeChooserEnvironment env, String[] options) {
-    if (!env.hasTableId())
+  public String choose(VolumeChooserEnvironment env, String[] options) throws AccumuloException {
+    if (!env.hasTableId() && !env.hasScope()) {
+      // this should only happen during initialize
+      log.warn("No table id or scope, so it's not possible to determine preferred volumes.  Using all volumes.");
       return super.choose(env, options);
+    }
+    ServerConfigurationFactory localConf = loadConf();
 
-    // Get the current table's properties, and find the preferred volumes property
-    // This local variable is an intentional component of the single-check idiom.
-    ServerConfigurationFactory localConf = serverConfs;
-    if (localConf == null) {
-      // If we're under contention when first getting here we'll throw away some initializations.
-      localConf = new ServerConfigurationFactory(HdfsZooInstance.getInstance());
-      serverConfs = localConf;
+    String systemPreferredVolumes = localConf.getConfiguration().get(PREFERRED_VOLUMES_CUSTOM_KEY);
+    if (null == systemPreferredVolumes || systemPreferredVolumes.isEmpty()) {
+      String logMessage = "Default preferred volumes are missing.";
+      log.debug(logMessage);
+      throw new AccumuloException(logMessage);
     }
-    TableConfiguration tableConf = localConf.getTableConfiguration(env.getTableId());
-    final Map<String,String> props = new HashMap<>();
-    tableConf.getProperties(props, PREFERRED_VOLUMES_FILTER);
-    if (props.isEmpty()) {
-      log.warn("No preferred volumes specified. Defaulting to randomly choosing from instance volumes");
-      return super.choose(env, options);
+
+    String volumes = null;
+    if (env.hasTableId()) {
+      volumes = localConf.getTableConfiguration(env.getTableId()).get(PREFERRED_VOLUMES_CUSTOM_KEY);
+    } else if (env.hasScope()) {
+      volumes = localConf.getConfiguration().get(Property.GENERAL_ARBITRARY_PROP_PREFIX.getKey() + env.getScope() + ".preferredVolumes");
```

Comment:
Would it make sense to add static that takes a scope argument and generated this property?  This could be a public static equivalent to `PREFERRED_VOLUMES_CUSTOM_KEY` for configuration purposes.